Expand documentation for log threshold alerts (#141)
elastic/observability-docs: docs/en/observability/logs-threshold-alert.asciidoc

There are several key supported use cases. You can create alerts based on fields containing or matching a text pattern,
alerts based on a numeric field and arithmetic operator, or a single alert with multiple conditions.

> Each of these comparators is backed by a different Elasticsearch query type, and in some scenarios it is important to know what these are so that you can configure your alert correctly. The comparators listed above map to the following Elasticsearch query types:

Reviewer: ```suggestion
A different {es} query type backs each of these comparators, and in some scenarios, it is important to know what these are so that you can configure your alert correctly. The comparators listed above map to the following {es} query types:
```

- *more than*: *range* using *gt*
- *more than or equals*: *range* using *gte*
- *less than*: *range* using *lt*
- *less than or equals*: *range* using *lte*

- *is* and *is not*: *term*

- *matches* and *does not match*: *match*

- *matches phrase* and *does not match phrase*: *match_phrase*

[discrete]
[[group-by]]
=== Group by

> It is possible to set a *group by* for log threshold alerts, you may set one or multiple groupings.

Reviewer: ```suggestion
It is possible to set a *group by* for log threshold alerts. You may set one or multiple groupings.
```

> In the scenario that a group by is set a composite aggregation will be performed against the selected group by fields. If any of these groups match the selected alert conditions, an alert will fire *per group*.

Reviewer: ```suggestion
When *group by* is set, a composite aggregation is performed against the selected fields. When any of these groups match the selected alert conditions, an alert fires *per group*.
```

> In scenarios where there are multiple groupings selected, the group name will be separated by commas.

Reviewer: ```suggestion
In scenarios where there are multiple groupings selected, the group name is separated by commas.
```

> For example if *host.name* and *host.architecture* were selected as group by fields, and there were two hosts (Host A and Host B) and two architectures (Architecture A and Architecture B), the composite aggregation would form multiple groups, we'll focus on the following two groups: *Host A, Architecture A* and *Host B, Architecture B*.

Reviewer: ```suggestion
For example, if *host.name* and *host.architecture* are selected as group by fields, and there are two hosts (Host A and Host B) and two architectures (Architecture A and Architecture B), the composite aggregation forms multiple groups. We'll focus on the *Host A, Architecture A* and *Host B, Architecture B* groups.
```

> If the group Host A, Architecture A matched the alert conditions, but Host B, Architecture B didn't, one alert would fire.

Reviewer: ```suggestion
If the group *Host A, Architecture A* matches the alert conditions, but *Host B, Architecture B* doesn't, one alert is triggered.
```

> Similarly, if there was a single group by selected (let's say *host.name*), and Host A matched the conditions, but Host B didn't, one alert would fire for Host A. If both groups matched the conditions, then two alerts would fire. 

Reviewer: ```suggestion
Similarly, if there was a single group by selected, for example, *host.name*, and Host A matches the conditions, but Host B doesn't, one alert is triggered for Host A. If both groups matches the conditions, then two alerts are triggered. 
```

> It's important to note that if group by fields are selected but there are no documents within the given time range (of alert execution) that contain the selected field(s) then we can't determine the group(s). This is relevant in scenarios where the alert condition is sensitive to there being a certain number of documents, and that number might be 0. To use a concrete example if we wanted to know that Host A had less than 5 documents matching a condition, but the host had stopped reporting logs for the entire duration we were querying, no alert would fire.

Reviewer: ```suggestion
[IMPORTANT]
=====
When group by fields are selected, but no documents are containing the selected field(s) within the given time range (of alert execution), then we can't determine the group(s). This is relevant when the alert condition is sensitive to a certain number of documents, and that number might be 0. To use an example, if we want to know that Host A has less than five documents matching a condition, but the host has stopped reporting logs for the entire duration we were querying, no alert is triggered.
=====
```

[role="screenshot"]
image::images/log-threshold-alert-group-by.png[Log threshold alert group by]

[discrete]
[[chart-previews]]
=== Chart previews

> It is possible to view a chart preview for each condition. This is useful for determining how many log entries would match each part of your configuration. When a group by is set, the chart will display a bar per group. This preview can be made visible by interacting with the arrow next to the condition.

Reviewer: ```suggestion
To determine how many log entries would match each part of your configuration, you can view a chart preview for each condition. This is useful for determining how many log entries would match each part of your configuration. When a group by is set, the chart displays a bar per group. To view the preview, select the arrow next to the condition.
```

image::images/log-threshold-alert-chart-previews.png[Log threshold chart previews]

The shaded area denotes the threshold that has been selected.

[discrete]
[[settings]]
=== Settings

> With log threshold alerts it's not possible to set an explicit index pattern as part of the configuration. The index pattern is instead inferred from the *Log indices* setting on the *Settings* page of the Logs application.

Reviewer: ```suggestion
With log threshold alerts, it's not possible to set an explicit index pattern as part of the configuration. The index pattern is instead inferred from  *Log indices* on the *Settings* page of the {logs-app}.
```

> It's important to note that this *Log indices* setting is looked up with each execution of the alert check, it is not stored at alert creation time.

Reviewer: ```suggestion
With each execution of the alert check, the *Log indices* setting is checked; however, it is not stored when the alert is created.
```

Extend your alerts by connecting them to actions that use the following supported built-in integrations.

[role="screenshot"]
image::images/action-type-logs.png[Alert action types]

[discrete]
[[es-queries]]
==== Elasticsearch queries (advanced)

> When an alert check is performed we build a query based on the alert configuration provided, for the vast majority of cases it shouldn't be necessary to know what these queries are. However, it might be useful to know the structure of these queries to determine an optimal configuration, or to aid with debugging. Below is an example Elasticsearch query for the following configuration:

Reviewer: ```suggestion
When an alert check is performed, a query is built based on the alert configuration. For the vast majority of cases it shouldn't be necessary to know what these queries are. However, to determine an optimal configuration or to aid with debugging, it might be useful to see the structure of these queries. Below is an example {es} query for the following configuration:
```
